### PR TITLE
JWKS for choreo branch

### PR DESCRIPTION
**Pull request by jhivandebenoit**
### Issues

Fixes 3148

https://github.com/wso2/product-microgateway/issues/3148

### Automation tests
 - Unit tests added: Yes
 - Integration tests added: Yes

### Tested environments
Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: adapter/config/default_config.go [hunk 1]
```diff
@@ -177,18 +177,23 @@ var defaultConfig = &Config{
...
+			EnableUserClaims:     false,
+			GatewayGeneratorImpl: "org.wso2.carbon.apimgt.common.gateway.jwtgenerator.APIMgtGatewayJWTGeneratorImpl",
+			ClaimsExtractorImpl:  "org.wso2.carbon.apimgt.impl.token.ExtendedDefaultClaimsRetriever",
+			TokenTTL:             3600,
+			Keypair: []KeyPair{
+				{
+					PublicCertificatePath: "/home/wso2/security/truststore/mg.pem",
+					PrivateKeyPath:        "/home/wso2/security/keystore/mg.key",
+					UseForSigning:         true,
+				},
```
Comment (pubudu538) [inline adapter/config/default_config.go:195]: We have to support the backward compatibility in this case. Basically existing configuration should work with this new implementation until we change the configurations.

Comment (jhivandebenoit) [reply adapter/config/default_config.go:195]: ack

**Comment by VirajSalaka** [comment]
We need to do filter out JWKS calls from being published to analytics, based on `/.wellknown/jwks` path. Let's do that modification in main branch too.

**Review by VirajSalaka — APPROVED** [review]

**Comment by choreo-cicd** [comment]
[failed] DataplaneE2 cluster : dev-deployment-v2 : 20221206.6

**Comment by choreo-cicd** [comment]
[failed] DataplaneE1 cluster : dev-deployment-v2 : 20221206.6

**Comment by choreo-cicd** [comment]
[failed]  : dev-deployment-v2 : 20221206.6

**Comment by choreo-cicd** [comment]
[failed] Controlplane cluster : dev-deployment-v2 : 20221206.6

**Comment by choreo-cicd** [comment]
[failed]  : dev-deployment-v2 : 20221206.7

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : stage-deployment-v2 : 20221208.2

**Comment by choreo-cicd** [comment]
[succeeded] DataplaneE1 cluster : stage-deployment-v2 : 20221208.2

**Comment by choreo-cicd** [comment]
[succeeded] DataplaneE2 cluster : dev-deployment-v2 : 20221208.14

**Comment by choreo-cicd** [comment]
[succeeded] DataplaneE1 cluster : dev-deployment-v2 : 20221208.14

**Comment by choreo-cicd** [comment]
[succeeded]  : dev-deployment-v2 : 20221208.14

**Comment by choreo-cicd** [comment]
[succeeded] DataplaneE1 cluster : prod-deployment-v2 : 20221208.5

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : prod-deployment-v2 : 20221208.5

**Pull request merged**